### PR TITLE
v1: wd_rsa.h remove local header path

### PR DESCRIPTION
**Pull request by gaozhangfei**
LIBRARY_PATH=/usr/local/lib C_INCLUDE_PATH=~/work/uadk/v1 make
uadk/v1/wd_rsa.h:23:19: fatal error: v1/wd.h: No such file or directory

Signed-off-by: Zhangfei Gao <zhangfei.gao@linaro.org>

### CODE REVIEW

**Comment by hzhuang1** [comment]
If user just build UADK v1, and install it into system directory. What's the instructions?

**Review by Liulongfang — COMMENTED** [review]
Please do not modify the behavior of our interface, if you want to adapt to openssl, please modify the middle layer of the kae engine

**Comment by gaozhangfei** [comment]
> Please do not modify the behavior of our interface, if you want to adapt to openssl, please modify the middle layer of the kae engine

Suppose this is port code error, could you help check with internal code.
Kae is already mature, but different behavior between uacce and warpdrive.
If change kae engine, internal warpdrive can not work

File: v1/wd_cipher.c [hunk 1]
```diff
@@ -385,7 +385,7 @@ static int cipher_recv_sync(struct wcrypto_cipher_ctx *ctx,
 		opdata[i]->status = resp[i]->result;
 	}
 
-	return recv_count;
+	return WD_SUCCESS;
```
Comment (Liulongfang) [inline v1/wd_cipher.c:388]: 这里不改，在下面这里改成这样：
int wcrypto_do_cipher(void *ctx, struct wcrypto_cipher_op_data *opdata,		void *tag)
{
        int ret;
        if (!tag)
                   ret = wcrypto_burst_cipher(ctx, &amp;opdata, NULL, 1);
        else
                   ret = wcrypto_burst_cipher(ctx, &amp;opdata, &amp;tag, 1);
        if (likely(ret == 1))
                   return WD_SUCCESS;
        return ret;
}

File: v1/wd_rsa.h [hunk 2]
```diff
@@ -20,7 +20,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <linux/types.h>
-#include "v1/wd.h"
+#include "wd.h"
```
Comment (Liulongfang) [inline v1/wd_rsa.h:24]: You should modify the way of external dependencies, directly changing the path inside will cause compilation errors

File: v1/wd_rsa.c [hunk 3]
```diff
@@ -923,11 +923,6 @@ static int rsa_request_init(struct wcrypto_rsa_msg *req, struct wcrypto_rsa_op_d
 			WD_ERR("sign or verf in_bytes != key_size!\n");
 			return -WD_EINVAL;
 		}
-
-		if (unlikely(req->out_bytes < c->key_size)) {
-			WD_ERR("out bytes %u error!\n", req->out_bytes);
-			return -WD_EINVAL;
-		}
 	}
```
Comment (Liulongfang) [inline v1/wd_rsa.c:926]: It may not work if the output length parameter is not set here。

File: v1/wd_cipher.c [hunk 4]
```diff
@@ -467,10 +467,17 @@ int wcrypto_burst_cipher(void *ctx, struct wcrypto_cipher_op_data **opdata,
...
+		ret = wcrypto_burst_cipher(ctx, &opdata, NULL, 1);
 	else
-		return wcrypto_burst_cipher(ctx, &opdata, &tag, 1);
+		ret = wcrypto_burst_cipher(ctx, &opdata, &tag, 1);
+
+	if (likely(ret == 1))
+		return WD_SUCCESS;
+
+	return ret;
 }
```
Comment (Liulongfang) [inline v1/wd_cipher.c:481]: Here is a complete modification method. I will send a patch tomorrow to fix this problem.

**Pull request closed without merging**